### PR TITLE
AdbProtocolHandler cleanup

### PR DESCRIPTION
**Pull request by janosvitok**
Some refactorings

### CODE REVIEW

**Comment by vidstige** [comment]
Why the "WIP" prefix? Should I review this? 🤔

**Comment by janosvitok** [comment]
I've rebased the PR and removed the WIP prefix. I will do the same with #89 after this is merged.

File: src/se/vidstige/jadb/server/AdbProtocolHandler.java [hunk 1]
```diff
@@ -37,91 +37,111 @@ public void run() {
...
-                }
-            } catch (ProtocolException e) {
-                output.writeBytes("FAIL");
-                send(output, e.getMessage());
+        try (
+            DataInputStream input = new DataInputStream(socket.getInputStream());
+            DataOutputStream output = new DataOutputStream(socket.getOutputStream())
+        ) {
+            while (processCommand(input, output)) {
+                // nothing to do here
```
Comment (vidstige) [inline src/se/vidstige/jadb/server/AdbProtocolHandler.java]: This seems weird. `processCommand` has side effects. It would be better to do

```java
while (true) {
    processCommand(input, output)
}
```

File: src/se/vidstige/jadb/server/AdbProtocolHandler.java [hunk 2]
```diff
@@ -37,91 +37,111 @@ public void run() {
...
+            } else if (command.startsWith("host-serial:")) {
+                hostSerial(output, command);
+            } else {
+                throw new ProtocolException("Unknown command: " + command);
+            }
+        } catch (ProtocolException e) {
+            output.writeBytes("FAIL");
+            send(output, e.getMessage());
+        }
+        output.flush();
```
Comment (vidstige) [inline src/se/vidstige/jadb/server/AdbProtocolHandler.java]: why is this needed?

File: src/se/vidstige/jadb/server/AdbProtocolHandler.java [hunk 3]
```diff
@@ -37,91 +37,111 @@ public void run() {
...
+        return true;
```
Comment (vidstige) [inline src/se/vidstige/jadb/server/AdbProtocolHandler.java]: always return `true` - Better to make this return void instead.

File: src/se/vidstige/jadb/server/AdbProtocolHandler.java (hunk 1 above)
Comment (janosvitok) [reply src/se/vidstige/jadb/server/AdbProtocolHandler.java]: processCommand needs to break the cycle.
Therefore:
```java
while (true) {
    if (!processCommand(...)) {
        break;
   }
}
```

File: src/se/vidstige/jadb/server/AdbProtocolHandler.java (hunk 2 above)
Comment (janosvitok) [reply src/se/vidstige/jadb/server/AdbProtocolHandler.java]: output.flush() was moved from another place (line 121 in current master)

File: src/se/vidstige/jadb/server/AdbProtocolHandler.java (hunk 3 above)
Comment (janosvitok) [reply src/se/vidstige/jadb/server/AdbProtocolHandler.java]: line 67 returns false in order to break the cycle in runServer().
false = stop the cycle
true = keep running

Maybe I should add boolean member stopServer, make processCommand return void, remove 'return true', replace 'return false' with 'stopServer = true' and change the cycle to while(!stopServer) { processCommand(); } ?

Another version:
```java
boolean keepRunning = true;
while (keepRunning) {
    keepRunning = processCommand(input, output);
}
```

What do you like best?

File: src/se/vidstige/jadb/server/AdbProtocolHandler.java (hunk 2 above)
Comment (vidstige) [reply src/se/vidstige/jadb/server/AdbProtocolHandler.java]: 👍

File: src/se/vidstige/jadb/server/AdbProtocolHandler.java (hunk 3 above)
Comment (vidstige) [reply src/se/vidstige/jadb/server/AdbProtocolHandler.java]: I see. No, then this is file. 👍
